### PR TITLE
ebmc: add API for ranking function checking

### PR DESCRIPTION
**Pull request by kroening**
This exposes two functions, `parse_ranking_function` and `is_ranking_function`, to enable the use of ranking function checking as a component.

### CODE REVIEW

File: src/ebmc/ranking_function.cpp [hunk 1]
```diff
@@ -114,51 +85,79 @@ ebmc_baset::propertyt &ranking_function_checkt::find_property()
...
 
   if(exit_code != -1)
     throw ebmc_errort();
 
   CHECK_RETURN(transition_system.trans_expr.has_value());
 
   // parse the ranking function
-  const auto ranking_function = parse_ranking_function();
+  if(!cmdline.isset("ranking-function"))
+    throw ebmc_errort() << "no candidate ranking function given";
```
Comment (tautschnig) [inline src/ebmc/ranking_function.cpp:105]: Is this really the right place for performing this check? I'd expect an invariant here for whatever code is to invoke `do_ranking_function` must only do so when this argument is present.

File: src/ebmc/ranking_function.cpp [hunk 2]
```diff
@@ -114,51 +85,79 @@ ebmc_baset::propertyt &ranking_function_checkt::find_property()
...
 
   // find the property
-  exit_code = get_properties();
+  ebmc_propertiest properties;
 
-  if(exit_code != -1)
-    return exit_code;
+  if(ebmc_propertiest::from_transition_system(
+       transition_system, properties, message_handler))
+    throw ebmc_errort();
```
Comment (tautschnig) [inline src/ebmc/ranking_function.cpp:115]: Is an exception (without explanation) the right thing here? Perhaps use `CHECK_RETURN` instead?

**Review by tautschnig — APPROVED** [review]

File: src/ebmc/ranking_function.cpp (hunk 2 above)
Comment (kroening) [reply src/ebmc/ranking_function.cpp:115]: I'll change `ebmc_propertiest::from_transition_system` to throw its own exceptions.

File: src/ebmc/ranking_function.cpp (hunk 1 above)
Comment (kroening) [reply src/ebmc/ranking_function.cpp:105]: That is `is_ranking_function` -- the code in `do_ranking_function` could move to ebmc_parse_options.cpp.

**Pull request merged**